File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/post/Reaction.kt
package com.adqmobile.socialnetwork.domain.post

import com.adqmobile.socialnetwork.data.remote.EnumByValueAdapter
import com.adqmobile.socialnetwork.data.remote.HasValue
import com.adqmobile.socialnetwork.domain.user.User
import com.google.gson.annotations.JsonAdapter

data class Reaction(val id: Long?,
                    val post: Post,
                    val reactionType: ReactionType,
                    val user: User,
                    val date: Long) {

    @JsonAdapter(EnumByValueAdapter::class)
    enum class ReactionType(override val value: String) : HasValue {
        REACTION_LIKE("like"),
        REACTION_DISLIKE("dislike")
    }
}

File: react-native/domain/post/Reaction.ts
import User from "../user/User";

export default class Reaction {
    postId: string;
    reactionType: string;
    user: User;
    date: number;

    constructor(postId: string, reactionType: string, user: User, date: number) {
        this.postId = postId;
        this.reactionType = reactionType;
        this.user = user;
        this.date = date;
    }
}

export class ReactionType {
    static REACTION_LIKE = "like";
    static REACTION_DISLIKE = "dislike";
}